pennylane/qchem/hamiltonian.py

```diff
@@ -107,6 +107,11 @@ def _electron_integrals(*args):
         Returns:
             tuple[array[float]]: 1D tuple containing core constant, one- and two-electron integrals
         """
+        if mol.n_electrons % 2 == 1 or mol.mult != 1:
+            raise ValueError(
+                "Open-shell systems are not supported. Change the charge or spin multiplicity of the molecule."
+            )
+
```

Comment:
Why this is needed in _electron_integrals? This function calls scf, which already has the check.